app/common/model/src/main/java/io/syndesis/common/model/bulletin/LeveledMessage.java

```diff
@@ -65,6 +65,8 @@ default Code getCode() {
 
     Optional<String> getMessage();
 
+    Optional<String> getDetail();
```

Comment:
@lburgazzoli 
Do you mean javadoc the code or outline it here?

Just in case ...
getDetail() is meant to be additional to both the code and message. Both of these have definite roles in the UI, ie. code translates to generic text and message is only displayed if one was specified.

Now as current design, if message is populated then notifications defer to displaying that message thereby rendering the UI defined generic messages pointless. Therefore, to preserve current functionality, detail is populated instead, allowing each LeveledMessage to contain a code and detail then display a generic message by the UI.

At this juncture, it was my preference not to disrupt the current UI handling of LeveledMessages hence the above implementation. Indeed, there is zero point until #3637 has come up with a new notification component and understand its design and population requirements.

Reply:
So the Equivalencer class replaces the equivalent() methods provided to each of the artifact interfaces, eg. Step, Integration etc... The tests conducted remain the same as before thereby if an object is found to be different the following kind of message is returned.

Template:
artifactName:artifactType > nestedArtifactName:nestedArtifactType > ... > ... > nestedArtifactName:nestedArtifactType$property-name
  => value of property from first object
  => value of property from second object

Example:
myIntegration:Integration > PostgresDB:Connection > sql:Connector > Invoke SQL:ConnectorAction$name
	=> 'Invoke SQL'
	=> 'Invoke PL-SQL'

So what it is saying is that the name property value of the 'SQL' ConnectionAction differs between 2 instances of the Integration 'myIntegration'

Therefore, in the context of the IntegrationUpdateHandler, an integration's nested connection is compared against the connection stored in the DataManager. At that point the 2 are compared for equivalence (this already happens) and rather than just a true/false result, messages such are the following would be returned:

PostgresDB:Connection > SQL Connector:Connector$name"
  => 'SQL Connec ... '
  => 'SQL Connec ... _new'

ie. the 'PostgresDB' Connection instance differ via their Connector name property due to the value having '_new' appended to the end of the second one.

Reply:
understood. I think for debugging purpose is ok but need to be revisited by UX